src/__tests__/WeekView.test.js

```diff
@@ -1,10 +1,446 @@
-import 'react-native';
 import React from 'react';
-import renderer from 'react-test-renderer';
+import { Dimensions } from 'react-native';
+import { render, cleanup } from '@testing-library/react-native';
+import { State } from 'react-native-gesture-handler';
+import {
+  fireGestureHandler,
+  getByGestureTestId,
+} from 'react-native-gesture-handler/jest-utils';
 import WeekView from '../WeekView/WeekView';
 
-it('renders correctly', () => {
-  renderer.create(
-    <WeekView numberOfDays={3} selectedDate={new Date(2021, 3, 24)} />,
-  );
+/**
+ * Extracts the event-id embedded in "WeekViewEvent-<id>".
+ *
+ * Throws an error if the testID does not match the regex.
+ * @param {String} testID
+ * @returns string with event id
+ */
+const extractEventIdFromTestID = (testID) => {
+  const found = testID.match(/WeekViewEvent-(?<eventId>\d+)/);
+  const eventId = found && found.groups && found.groups.eventId;
+  if (!eventId) {
+    throw Error(`Could not find the event id in testID: ${testID}`);
+  }
+  return eventId;
+};
+
+/**
+ * NOTE: In most cases the props provided to <WeekView /> need to be
+ * selected carefully, so that events _are visible in the screen_.
+ * For example:
+ *   - numberOfDays
+ *   - selectedDate (usually, at least the same day the target event's startDate)
+ *   - hoursInDisplay
+ *   - startHour (usually, some hours before the target event)
+ */
+describe('Basic render functionality', () => {
+  it('Renders with no events', () => {
+    // does not throw exception
+    render(
+      <WeekView
+        events={[]}
+        numberOfDays={3}
+        selectedDate={new Date(2020, 3, 24)}
+      />,
+    );
+  });
+
+  const runTestShowsAll = (baseEvents, baseProps) => {
+    const events = baseEvents;
+    const { queryAllByHintText } = render(
+      /* eslint-disable-next-line react/jsx-props-no-spreading */
+      <WeekView events={events} {...baseProps} />,
+    );
+    /**
+     * NOTE: Ideally, the selection would be by text: queryByText('event something')
+     * instead of queryByHintText(), but in practice the event description might be
+     * hidden due to the size of the events --> queryByText() does not find it, even though
+     * the event is visible in the screen as a box with a color.
+     * (Is hard to ensure that the event size is large enough to display the description:
+     * depends on the screen size, hours in display, number of days, font size, padding, etc.)
+     */
+    expect(queryAllByHintText('Show event')).toBeArrayOfSize(events.length);
```

Comment:
if we want to test if it renders events correctly, I think we should grab the parent container or use the regex pattern

Reply:
I changed for the following: using byHint first and then byText.

* If both tests fail --> events are not rendered correctly --> something is broken
* If both tests pass --> events are being rendered correctly
* If byHint passes and byText fails --> boxes are rendered but text is hidden --> events are too small

i.e. I once got stuck with this for hours, byText() was failing and could not find why. Then I realized the code was not broken, the events were too small :sweat_smile: